src/routers/alpha-router/gas-models/gas-model.ts

```diff
@@ -78,6 +79,24 @@ export type L1ToL2GasCosts = {
   gasCostL1QuoteToken: CurrencyAmount;
 };
 
+export type BuildOnChainGasModelFactoryType = {
+  chainId: ChainId;
+  gasPriceWei: BigNumber;
+  V3poolProvider: IV3PoolProvider;
+  token: Token;
+  l2GasDataProvider?:
+    | IL2GasDataProvider<OptimismGasData>
+    | IL2GasDataProvider<ArbitrumGasData>;
+  V2poolProvider?: IV2PoolProvider;
```

Comment:
Whys this optional? 

Even if its not used in some cases I'd just make it required, and just not use it in the implementation that doesn't need it

Reply:
Sounds good